From 2d3d009bf4552ae3e25c5a48381bedf7b57642d6 Mon Sep 17 00:00:00 2001
From: Yong Yao <yong.yao@intel.com>
Date: Wed, 25 Sep 2013 09:40:15 +0800
Subject: Revert "Make ctype.h a little less unhygienic."

BZ: 149724

This reverts commit 3912f90e56780ef54c68c796ff6793f5692e7658.

Change-Id: Ib2a3fe0110412c27596df8766eb53e787ae08043
Category: device enablement
Domain: AOSP-bionic
Origin: internal
Upstream-Candidate: no
---
 libc/include/ctype.h |   38 +++++++++++++++++++-------------------
 libc/stdlib/ctype_.c |   12 +-----------
 2 files changed, 20 insertions(+), 30 deletions(-)

diff --git a/libc/include/ctype.h b/libc/include/ctype.h
index 5557e31..58b76ea 100644
--- a/libc/include/ctype.h
+++ b/libc/include/ctype.h
@@ -42,14 +42,14 @@
 
 #include <sys/cdefs.h>
 
-#define	_CTYPE_U	0x01
-#define	_CTYPE_L	0x02
-#define	_CTYPE_N	0x04
-#define	_CTYPE_S	0x08
-#define	_CTYPE_P	0x10
-#define	_CTYPE_C	0x20
-#define	_CTYPE_X	0x40
-#define	_CTYPE_B	0x80
+#define	_U	0x01
+#define	_L	0x02
+#define	_N	0x04
+#define	_S	0x08
+#define	_P	0x10
+#define	_C	0x20
+#define	_X	0x40
+#define	_B	0x80
 
 __BEGIN_DECLS
 
@@ -101,57 +101,57 @@ int	_toupper(int);
 
 __CTYPE_INLINE int isalnum(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_CTYPE_U|_CTYPE_L|_CTYPE_N)));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_U|_L|_N)));
 }
 
 __CTYPE_INLINE int isalpha(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_CTYPE_U|_CTYPE_L)));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_U|_L)));
 }
 
 __CTYPE_INLINE int iscntrl(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_C));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _C));
 }
 
 __CTYPE_INLINE int isdigit(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_N));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _N));
 }
 
 __CTYPE_INLINE int isgraph(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_CTYPE_P|_CTYPE_U|_CTYPE_L|_CTYPE_N)));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_P|_U|_L|_N)));
 }
 
 __CTYPE_INLINE int islower(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_L));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _L));
 }
 
 __CTYPE_INLINE int isprint(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_CTYPE_P|_CTYPE_U|_CTYPE_L|_CTYPE_N|_CTYPE_B)));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_P|_U|_L|_N|_B)));
 }
 
 __CTYPE_INLINE int ispunct(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_P));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _P));
 }
 
 __CTYPE_INLINE int isspace(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_S));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _S));
 }
 
 __CTYPE_INLINE int isupper(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _CTYPE_U));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & _U));
 }
 
 __CTYPE_INLINE int isxdigit(int c)
 {
-	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_CTYPE_N|_CTYPE_X)));
+	return (c == -1 ? 0 : ((_ctype_ + 1)[(unsigned char)c] & (_N|_X)));
 }
 
 __CTYPE_INLINE int tolower(int c)
diff --git a/libc/stdlib/ctype_.c b/libc/stdlib/ctype_.c
index 3703f64..cf32f16 100644
--- a/libc/stdlib/ctype_.c
+++ b/libc/stdlib/ctype_.c
@@ -36,15 +36,6 @@
 #include <ctype.h>
 #include "ctype_private.h"
 
-#define _U _CTYPE_U
-#define _L _CTYPE_L
-#define _N _CTYPE_N
-#define _S _CTYPE_S
-#define _P _CTYPE_P
-#define _C _CTYPE_C
-#define _X _CTYPE_X
-#define _B _CTYPE_B
-
 const char _C_ctype_[1 + CTYPE_NUM_CHARS] = {
 	0,
 	_C,	_C,	_C,	_C,	_C,	_C,	_C,	_C,
@@ -86,8 +77,6 @@ const char _C_ctype_[1 + CTYPE_NUM_CHARS] = {
 const char *_ctype_ = _C_ctype_;
 
 
-// TODO: fix the header file so we don't have to duplicate all this inlined stuff.
-
 #if 1 /* ndef NDEBUG */
 int isalnum(int c)
 {
@@ -166,3 +155,4 @@ int toascii(int c)
 #endif /* __BSD_VISIBLE || __XPG_VISIBLE */
 
 #endif /* !NDBEUG */
+
-- 
1.7.4.1

